lib/signalfx.js

```diff
@@ -0,0 +1,334 @@
+    } else if (typeof datapoint.value === 'number' && datapoint.value % 1 === 0) { // int value
+      protobufDatapoint.value.intValue = Math.round(+datapoint.value);
+    } else {
+      throw new TypeError('Invalid Value ' + datapoint.value);
+    }
+
+    protobufDatapoint.metricType = protocolBuffers.MetricType[metricType.toUpperCase()];
+    protobufDatapoint.metric = datapoint['metric'];
+
+    if (datapoint['timestamp']) {
+      protobufDatapoint.timestamp = +datapoint['timestamp'];
+    }
+
+    // set datapoint dimensions
+    var dimensions = datapoint['dimensions'] || [];
+    for (var key in dimensions) {
+      if (dimensions.hasOwnProperty(key)) {
+        var dim = {};
+        dim.key = key;
+        dim.value = dimensions[key];
+        protobufDatapoint.dimensions.push(dim);
+      }
+    }
+    this.queue.push(protobufDatapoint);
+  };
+
+  ProtoBufSignalFx.prototype.batchData = function (datapointsList) {
+    var dpum = new protocolBuffers.DataPointUploadMessage();
+    dpum.datapoints = dpum.datapoints.concat(datapointsList);
+
+    try {
+      var dataToSend = dpum.encode();
+    } catch (error) {
+      winston.error('Invalid Protobuf object', error);
+    }
+    return dataToSend.toBuffer();
+  };
+
+
+// ============================================================================================
+//                  SignalFx API client with JSON data sending
+// ============================================================================================
+  /**
+   * SignalFx API client data handler that uses Json
+   * This class presents the interfaces that handle the serialization of data
+   * using Json
+   *
+   * @param apiToken
+   * @constructor
+   */
+  function JsonSignalFx(apiToken) {
+    SignalFxClient.apply(this, arguments);
+  }
+
+  JsonSignalFx.prototype = Object.create(SignalFxClient.prototype);
+  JsonSignalFx.prototype.constructor = SignalFxClient;
+
+  JsonSignalFx.prototype.getHeaderContentType = function () {
+    return JSON_HEADER_CONTENT_TYPE;
+  };
+
+  JsonSignalFx.prototype.addToQueue = function (metricType, datapoint) {
+    var jsonObject = {};
+    jsonObject[metricType] = datapoint;
+    this.queue.push(jsonObject);
+  };
+
+  JsonSignalFx.prototype.batchData = function (datapointsList) {
+    var data = {};
+    datapointsList.forEach(function (element) {
+
+      for (var metricType in element) {
+        if (element.hasOwnProperty(metricType)) {
+          if (!data[metricType]) {
+            data[metricType] = [];
+          }
+          data[metricType].push(element[metricType]);
+        }
+      }
+    });
+    return JSON.stringify(data);
+  };
+
+// ============================================================================================
+//                                 Export modules
+// ============================================================================================
+  var exports = module.exports = {};
```

Comment:
This line isn't necessary, module.exports is already an object, and there already exists a module-level var called exports which you can assign properties to.